src/core/server/saved_objects/import/extract_errors.ts

```diff
@@ -16,18 +16,47 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import { SavedObjectsClientContract } from 'src/core/server';
 import { SavedObject } from '../types';
 import { SavedObjectsImportError } from './types';
 
-export function extractErrors(
+export async function extractErrors(
   savedObjectResults: SavedObject[],
-  savedObjectsToImport: SavedObject[]
+  savedObjectsToImport: SavedObject[],
+  savedObjectsClient: SavedObjectsClientContract,
+  namespace?: string
 ) {
   const errors: SavedObjectsImportError[] = [];
   const originalSavedObjectsMap = new Map<string, SavedObject>();
+  const savedObjectsTitlesMap = new Map<string, string>();
+
   for (const savedObject of savedObjectsToImport) {
     originalSavedObjectsMap.set(`${savedObject.type}:${savedObject.id}`, savedObject);
   }
+
+  const errorSavedObjects = new Array();
+  for (const savedObject of savedObjectResults) {
+    if (savedObject.error) {
+      errorSavedObjects.push(savedObject);
+    }
+  }
+
+  // return an empty array if there are no errors
+  if (errorSavedObjects.length === 0) {
+    return [];
+  }
```

Comment:
Can we test for `savedObject.error.statusCode === 409` instead of just `savedObject.error`.

Nit: This code would be much more compact and readable if you used an array.filter like:
```ts
const errorSavedObjects = savedObjectResults.filter((so) => so.error && so.error.statusCode === 409)
```